build.gradle.kts

```diff
@@ -54,19 +54,62 @@ tasks {
 
     javadoc {
         // Suppressing Javadoc warnings is clunky, but there doesn't seem to be any nicer way to do it.
-        // https://stackoverflow.com/questions/52205209/configure-gradle-build-to-suppress-javadoc-console-warnings
+        // https://stackoverflow.com/questions/62894307/option-xdoclintnone-does-not-work-in-gradle-to-ignore-javadoc-warnings
         options {
-            (this as CoreJavadocOptions).addStringOption("Xdoclint:none", "-quiet")
+            this as StandardJavadocDocletOptions
+            addBooleanOption("Xdoclint:none", true)
+            addStringOption("Xmaxwarns", "1") // best we can do is limit warnings to 1
+        }
+    }
+
+    // spotbugs-gradle-plugin creates a :spotbugsTest task by default, but we don't want it
+    // see: https://github.com/spotbugs/spotbugs-gradle-plugin/issues/391
+    project.gradle.startParameter.excludedTaskNames.add(":spotbugsTest")
+
+    spotbugsMain {
+        val spotbugsBaselineFile = "$rootDir/config/spotbugs/baseline.xml"
+
+        if (!project.hasProperty("baseline")) { // the common case
+            baselineFile.set(file(spotbugsBaselineFile))
+            reports.create("html") {
+                required.set(true)
+            }
+        } else { // Get here via e.g. `./gradlew :spotbugsMain -Pbaseline`
+            // Note this path succeeds :spotbugsMain because *of course it does*
+            ignoreFailures = true
+            reports.create("xml") {
+                // Why bother? Otherwise we have kilobytes of workspace-specific absolute paths, statistics, etc.
+                // cluttering up the baseline XML and preserved in perpetuity. It would be far better if we could use
+                // the SpotBugs relative path support, but the way SpotBugs reporters are presently architected they
+                // drop the `destination` information long before Project.writeXML uses its presence/absence to
+                // determine whether to generate relative instead of absolute paths. So, contribute patch to SpotBugs or
+                // write own SpotBugs reporter in parallel or... do this.
+                // Improvements are definitely possible, but left as an exercise to the reader.
+                doLast {
+                    // It would be super neat if we had some way to handle this xml processing directly inline, without
+                    // generating a temp file or at least without shelling out.
+                    // Use ant's xslt capabilities? Xalan? Saxon gradle plugin (eerohele)? javax.xml.transform?
+                    // In the mean time... xsltproc!
+                    exec {
+                        commandLine(
+                            "xsltproc",
+                            "--output", spotbugsBaselineFile,
+                            "$rootDir/config/spotbugs/baseline.xslt",
+                            outputLocation.get().toString()
+                        )
+                    }
```

Comment:
I wanted to add one more line here:
```kotlin
throw Exception("You regenerated the SpotBugs baseline. You lose, sir! You get NOTHING!")
```
... causing a build that regenerates the baseline to be a failure. But then I realized that's pretty counter-productive. 🤷🏻‍♂️

Reply:
I mean... it could be useful if your goal is to make sure that the baseline task cannot be used in a CI workflow.